src/description/description/common.xacro

```diff
@@ -0,0 +1,232 @@
+<?xml version="1.0"?>
+<robot xmlns:xacro="http://www.ros.org/wiki/xacro">
+  <!-- Colours -->
+    <material name="white">
+        <color rgba="1 1 1 1"/>
+    </material>
+
+    <material name="orange">
+        <color rgba="1 0.3 0.1 1"/>
+    </material>
+
+    <material name="blue">
+        <color rgba="0.2 0.2 1 1"/>
+    </material>
+
+    <material name="black">
+        <color rgba="0 0 0 1"/>
+    </material>
+
+    <material name="yellow">
+        <color rgba="1.0 1.0 0.0 1.0"/>
+    </material>
+
+    <material name="green">
+        <color rgba="0.0 1.0 0.0 1.0"/>
+    </material>
+
+  <!-- Properties -->
+    <!-- Stepper Motor Properties -->
+        <xacro:property name="stepper_min_angle" value="0"/>
+        <xacro:property name="stepper_max_angle" value="6.28"/>
+
+    <!-- Carriage Properties -->
+        <xacro:property name="carriage_min_position" value="-0.68"/>
+        <xacro:property name="carriage_max_position" value="0.68"/>
+        <xacro:property name="carriage_stepper_offset" value="${carriage_max_position}"/> <!-- Assuming a min angle of 0 rad for the stepper -->
+        <xacro:property name="carriage_stepper_multiplier" value="${(carriage_min_position - carriage_stepper_offset) / stepper_max_angle}"/>
```

Comment:
1. You are correct, and this is intentional, because I wanted the carriage to start from close to its stepper motor and move to the other stepper motor as the stepper revolves.

2. I am not completely sure if I understand what you are saying here. However, this motion translation/mimicking can be tuned in the future no problem, I am not worried about that or any similar tuning.

I have used passed arguments in the URDF before so we can do that no problem in the next phase, and those parameters will be passed to the URDF in the launch file or set as constants somewhere else.